lib/lotus/model/adapters/sql/command.rb

```diff
@@ -55,10 +59,41 @@ def update(entity)
           # @api private
           # @since 0.1.0
           def delete
-            @collection.delete
+            _rescue_database_error do
+              @collection.delete
+            end
           end
 
           alias_method :clear, :delete
+
+          private
+
+          # Rescues Sequel::DatabaseError in yielded block
+          #
+          # @api private
+          def _rescue_database_error
+            yield
+          rescue Sequel::DatabaseError => e
+            raise _transform_database_error(e)
```

Comment:
However, not providing an abstracted error means that applications will have to be coupled directly to the adapter-specific error.  For example, if the application wanted to do something when an entity is not persisted because of a uniqueness constraint, the application would have to inspect the `Lotus::Model::Error#cause` for `Sequel::UniqueConstraintViolationError`.  This seems to defeat the purpose of the adapter abstraction.  What do you suggest applications do in such cases?